CodeGeneration/Sources/SyntaxSupport/Traits.swift

```diff
@@ -14,12 +14,14 @@ import SwiftSyntax
 
 public class Trait {
   public let traitName: String
+  public let syntaxBaseName: TokenSyntax?
```

Comment:
I would change this to `baseKind: SyntaxNodeKind?` and then use `baseKind.protocolType` to get the type that you want to get the protocol’s type name. 

Also: A doc comment would be appreciated.

Reply:
Should be fixed in [1166176](https://github.com/apple/swift-syntax/pull/2366/commits/11661761d2693e2a80bdec8b12f089e511a54a3f)